### PR TITLE
have createSdkContext create sdkPackage, createTcgcContext not

### PR DESCRIPTION
**Pull request by iscai-msft**

### CODE REVIEW

File: packages/typespec-autorest-canonical/src/emitter.ts [hunk 1]
```diff
@@ -27,6 +27,7 @@ import {
   getVersion,
 } from "@typespec/versioning";
 import { AutorestCanonicalEmitterOptions, reportDiagnostic } from "./lib.js";
+import { TCGCContext } from "../../typespec-client-generator-core/dist/src/internal-utils.js";
```
Comment (timotheeguerin) [inline packages/typespec-autorest-canonical/src/emitter.ts:30]: this also need to be exported

File: packages/typespec-autorest/src/utils.ts [hunk 2]
```diff
@@ -14,12 +14,13 @@ import {
 } from "@typespec/compiler";
 import { getOperationId } from "@typespec/openapi";
 import { pascalCase } from "change-case";
+import { TCGCContext } from "../../typespec-client-generator-core/dist/src/internal-utils.js";
```
Comment (timotheeguerin) [inline packages/typespec-autorest/src/utils.ts:17]: same here

File: packages/typespec-client-generator-core/src/interfaces.ts [hunk 3]
```diff
@@ -18,7 +22,38 @@ import {
   HttpVerb,
   Visibility,
 } from "@typespec/http";
-import { TCGCContext } from "./internal-utils.js";
+import { TspLiteralType } from "./internal-utils.js";
+
+export interface TCGCContext {
```
Comment (timotheeguerin) [inline packages/typespec-client-generator-core/src/interfaces.ts:27]: is there too many props here? I don't see them set in the createTCGCcontecxt?

File: packages/typespec-client-generator-core/src/interfaces.ts [hunk 4]
```diff
@@ -18,7 +22,38 @@ import {
   HttpVerb,
   Visibility,
 } from "@typespec/http";
-import { TCGCContext } from "./internal-utils.js";
```
Comment (timotheeguerin) [inline packages/typespec-client-generator-core/src/interfaces.ts:21]: I don't see the other one being removed

File: packages/typespec-client-generator-core/src/interfaces.ts (hunk 3 above)
Comment (iscai-msft) [reply packages/typespec-client-generator-core/src/interfaces.ts:27]: it currently is only adding the required props, i have optional props that are more for creating the sdk package, so I'm trying to have `createTcgcContext` create the minimal interface needed

File: packages/typespec-client-generator-core/src/validate.ts [hunk 5]
```diff
@@ -16,12 +16,13 @@ import {
...
-import { getClientNameOverride } from "./decorators.js";
-import { AllScopes, clientNameKey, createTCGCContext, TCGCContext } from "./internal-utils.js";
+import { getClientNameOverride, createTCGCContext } from "./decorators.js";
+import { AllScopes, clientNameKey } from "./internal-utils.js";
 import { reportDiagnostic } from "./lib.js";
+import { TCGCContext } from "./interfaces.js";
 
 export function $onValidate(program: Program) {
-  const tcgcContext = createTCGCContext(program);
+  const tcgcContext = createTCGCContext(program, "@azure-tools/typespec-client-generator-core");
```
Comment (timotheeguerin) [inline packages/typespec-client-generator-core/src/validate.ts:25]: I feel like it is good that this was not provided

File: packages/typespec-client-generator-core/src/interfaces.ts (hunk 3 above)
Comment (timotheeguerin) [reply packages/typespec-client-generator-core/src/interfaces.ts:27]: but wouldn't you want that on the `SDKContext` then?

File: packages/typespec-autorest/src/utils.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { SdkContext, getClientNameOverride } from "@azure-tools/typespec-client-generator-core";
+import { getClientNameOverride, TCGCContext } from "@azure-tools/typespec-client-generator-core";
```
Comment (timotheeguerin) [inline packages/typespec-autorest/src/utils.ts:1]:
```suggestion
import { getClientNameOverride, type TCGCContext } from "@azure-tools/typespec-client-generator-core";
```

File: packages/typespec-client-generator-core/src/interfaces.ts (hunk 3 above)
Comment (iscai-msft) [reply packages/typespec-client-generator-core/src/interfaces.ts:27]: ok that's fair, let me go ahead and do that. thanks!

File: packages/typespec-client-generator-core/src/validate.ts (hunk 5 above)
Comment (iscai-msft) [reply packages/typespec-client-generator-core/src/validate.ts:25]: why would it be good if it wasn't applied?

Comment (timotheeguerin) [reply packages/typespec-client-generator-core/src/validate.ts:25]: We are technically in a stage where there is no specific scope so this is kinda lying and might have unexpected consequences

**Comment by iscai-msft** [comment]
@timotheeguerin, circling back to the optional params in the SdkContext interface, they are actually needed in `getAllModels`, which TCGCContext needs to be able to be passed to (even though we aren't automatically calling it in `createTcgcContext`. The reason they're optional is because they're created on the fly. I've moved them back to `TcgcContext`, let me know if there's a better typescript design here that could happen though

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**